ote_cli/ote_cli/tools/demo.py

```diff
@@ -0,0 +1,202 @@
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import argparse
+import time
+
+import cv2
+import numpy as np
+from ote_sdk.configuration.helper import create
+from ote_sdk.entities.annotation import AnnotationSceneEntity, AnnotationSceneKind
+from ote_sdk.entities.datasets import DatasetEntity, DatasetItemEntity
+from ote_sdk.entities.id import ID
+from ote_sdk.entities.image import Image
+from ote_sdk.entities.inference_parameters import InferenceParameters
+from ote_sdk.entities.label import LabelEntity
+from ote_sdk.entities.label_schema import LabelSchemaEntity
+from ote_sdk.entities.model import ModelEntity
+from ote_sdk.entities.task_environment import TaskEnvironment
+from ote_sdk.usecases.adapters.model_adapter import ModelAdapter
+
+from ote_cli.registry import find_and_parse_model_template
+from ote_cli.tools.utils.demo.images_capture import open_images_capture
+from ote_cli.tools.utils.demo.visualization import draw_predictions, put_text_on_rect_bg
+from ote_cli.utils.config import override_parameters
+from ote_cli.utils.importing import get_impl_class
+from ote_cli.utils.loading import load_model_weights
+from ote_cli.utils.parser import (
+    add_hyper_parameters_sub_parser,
+    gen_params_dict_from_args,
+)
+
+ESC_BUTTON = 27
+
+
+def parse_args():
+    """
+    Parses command line arguments.
+    """
+
+    pre_parser = argparse.ArgumentParser(add_help=False)
+    pre_parser.add_argument("template")
+    parsed, _ = pre_parser.parse_known_args()
+    # Load template.yaml file.
+    template = find_and_parse_model_template(parsed.template)
+    # Get hyper parameters schema.
+    hyper_parameters = template.hyper_parameters.data
+    assert hyper_parameters
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("template")
+    parser.add_argument(
+        "-i",
+        "--input",
+        required=True,
+        help="Source of input data: images folder, image, webcam.",
+    )
+    parser.add_argument(
+        "--load-weights",
+        required=True,
+        help="Load only weights from previously saved checkpoint",
+    )
+    parser.add_argument(
+        "--fit-to-size",
+        nargs=2,
+        type=int,
+        help="Width and Height space-separated values. "
+        "Fits displayed images to window with specified Width and Height. "
+        "This options applies to result visualisation only.",
+    )
+    parser.add_argument(
+        "--loop", action="store_true", help="Enable reading the input in a loop."
+    )
+    parser.add_argument(
+        "--delay", type=int, default=3, help="Frame visualization time in ms."
+    )
+    parser.add_argument(
+        "--labels", nargs="+", required=True, help="A space-separated labels list."
```

Comment:
IMO, it would be good to describe all required arguments first and then the rest.

Reply:
And what about saving/loading the labels as a part persistent model representation?

Reply:
oops